RFS/src/main/java/org/opensearch/migrations/bulkload/workcoordination/OpenSearchWorkCoordinator.java

```diff
@@ -645,6 +665,145 @@ private WorkItemAndDuration getAssignedWorkItem(LeaseChecker leaseChecker,
         }
     }
 
+    private void updateWorkItemWithSuccessors(String workItemId, ArrayList<String> successorWorkItemIds) throws IOException {
+        final var updateSuccessorWorkItemsTemplate = "{\n"
+                + "  \"script\": {\n"
+                + "    \"lang\": \"painless\",\n"
+                + "    \"params\": { \n"
+                + "      \"clientTimestamp\": " + CLIENT_TIMESTAMP_TEMPLATE + ",\n"
+                + "      \"workerId\": \"" + WORKER_ID_TEMPLATE + "\",\n"
+                + "      \"successorWorkItems\": \"" + SUCCESSOR_WORK_ITEM_IDS_TEMPLATE + "\"\n"
+                + "    },\n"
+                + "    \"source\": \""
+                + "      if (ctx._source.scriptVersion != \\\"" + SCRIPT_VERSION_TEMPLATE + "\\\") {"
+                + "        throw new IllegalArgumentException(\\\"scriptVersion mismatch.  Not all participants are using the same script: sourceVersion=\\\" + ctx.source.scriptVersion);"
+                + "      }"
+                + "      if (ctx._source." + LEASE_HOLDER_ID_FIELD_NAME + " != params.workerId) {"
+                + "        throw new IllegalArgumentException(\\\"work item was owned by \\\" + ctx._source."
+                +                        LEASE_HOLDER_ID_FIELD_NAME + " + \\\" not \\\" + params.workerId);"
+                + "      }"
```

Comment:
Discussed with Greg -- because of the extra check on line 685, there's a risk that this could be picked up again, _but_ it will never cause this to spiral out of control. The second worker to grab it will never be able to create its own successor work items

Reply:
There may be a race condition in all of these calls since our final write isn't confirming that the document that we're writing was the same version.  I don't expect that we'll be using this datastore for the long term to do transactional coordination.  As Mikayla points out, this is a bounded efficiency problem and won't affect correctness, so this seems like a calculated risk that isn't worth putting more effort into at this point.